Makefile

```diff
@@ -138,39 +162,57 @@ agent: cmd/agent/agent
 agentctl: cmd/agentctl/agentctl
 agent-operator: cmd/agent-operator/agent-operator
 
-cmd/agent/agent: check-seego cmd/agent/main.go
+
+
+
+# In general DRONE variable should overwrite any other options, if DRONE is not set then fallback to normal behavior
+
+cmd/agent/agent: check-seego  cmd/agent/main.go
+ifeq ($(DRONE),true)
+	$(CGO_CC_CCX); go build $(CGO_FLAGS) -o $@ ./$(@D)
+else
 ifeq ($(CROSS_BUILD),false)
 	CGO_ENABLED=1 go build $(CGO_FLAGS) -o $@ ./$(@D)
 else
-	@CGO_ENABLED=1 GOOS=$(GOOS) GOARCH=$(GOARCH) GOARM=$(GOARM); $(seego) build $(CGO_FLAGS) -o $@ ./$(@D)
+	CGO_ENABLED=1 GOOS=$(GOOS) GOARCH=$(GOARCH) GOARM=$(GOARM); $(seego) build $(CGO_FLAGS) -o $@ ./$(@D)
+endif
 endif
 	$(NETGO_CHECK)
 
+
 cmd/agentctl/agentctl: check-seego cmd/agentctl/main.go
+ifeq ($(DRONE),true)
+	$(CGO_CC_CCX); go build $(CGO_FLAGS) -o $@ ./$(@D)
```

Comment:
This can use the seego entrypoint script instead of having to set CGO_CC_CXX here.

Reply:
Since we already have to define the GOOS, GOARCH, GOARM  including the CC CCX ect seems easier than calling out to an external script. Less context switching than going to the seego repo and looking at the go_wrapper.sh there. Another alternative would be making seego/go_wrapper aware of TARGETPLATFORM. Though I think that feels like even more indirection.

Reply:
> Another alternative would be making seego/go_wrapper aware of TARGETPLATFORM. 

Isn't that what https://github.com/grafana/agent/blob/main/tools/cross_build.bash does? 

> Since we already have to define the GOOS, GOARCH, GOARM including the CC CCX ect seems easier than calling out to an external script.

🤷‍♂️ seego is designed to be an abstraction over the go binary, and there's no guarantee that the same CC/CXX/etc values will be used between seego releases. I really think we should just rely on having seego figure it out.

Reply:
As previously discussed rolled back to using the bash file or default go build.